algorithms/kernel/svm/oneapi/svm_train_thunder_oneapi_impl.i

```diff
@@ -0,0 +1,298 @@
+/* file: svm_train_thunder_oneapi_impl.i */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  SVM training algorithm implementation thunder method
+//--
+*/
+/*
+//  DESCRIPTION
+//
+//  Definition of the functions for training with SVM 2-class classifier.
+//
+//  REFERENCES
+//
+
+//  1. Zeyi Wen, Jiashuai Shi, Bingsheng He
+//     ThunderSVM: A Fast SVM Library on GPUs and CPUs,
+//     Journal of Machine Learning Research, 19, 1-5 (2018)
+//  2. Rong-En Fan, Pai-Hsuen Chen, Chih-Jen Lin,
+//     Working Set Selection Using Second Order Information
+//     for Training Support Vector Machines,
+//     Journal of Machine Learning Research 6 (2005), pp. 1889___1918
+//  3. Bernard E. boser, Isabelle M. Guyon, Vladimir N. Vapnik,
+//     A Training Algorithm for Optimal Margin Classifiers.
+//  4. Thorsten Joachims, Making Large-Scale SVM Learning Practical,
+//     Advances in Kernel Methods - Support Vector Learning
+*/
```

Comment:
Are we allowed to have such references in our code? @rlnx @averbukh

Reply:
it's a good practice, we have this in some C++ algorithms already. It really helps to understand an algorithm.